### PR TITLE
Added MetaMask version check

### PR DESCRIPTION
**Pull request by WhiteNik16**

### CODE REVIEW

File: packages/connector/src/consts/general.ts [hunk 1]
```diff
@@ -1 +1,2 @@
 export const CORE_POLLING_INTERVAL = 5_000;
+export const METAMASK_SNAPS_MAJOR_VERSION = 11;
```
Comment (ihordiachenko) [inline packages/connector/src/consts/general.ts]: Checking the major version is not enough. Let's use something like this to compare with the required version:
https://www.npmjs.com/package/compare-versions

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - `@rarimo/rarime` - credentials storage tests
 - `@rarimo/rarime` - add RariMe App hostnames to the get credentials whitelist
+- `@rarimo/rarime` - Checking MetaMask version support
```
Comment (ihordiachenko) [inline CHANGELOG.md]:
```suggestion
- `@rarimo/rarime` - Checking MetaMask version compatibility
```

File: packages/connector/src/utils.ts [hunk 3]
```diff
@@ -76,3 +77,13 @@ export const isSnapInstalled = async (
     return false;
   }
 };
+
+export const checkSnapSupport = async () => {
+  const version = await window?.ethereum?.request?.({
+    method: 'web3_clientVersion',
+  });
+
+  const majorVersion = Number(version.match(/.+v([0-9]+).+/u)?.[1] ?? 0);
```
Comment (ardier16) [inline packages/connector/src/utils.ts]: seems like version could be undefined here

File: packages/connector/src/utils.ts [hunk 4]
```diff
@@ -76,3 +78,15 @@ export const isSnapInstalled = async (
...
+
+export const checkSnapSupport = async () => {
+  const version = await window?.ethereum?.request?.({
+    method: 'web3_clientVersion',
+  });
+
+  const currentVersion = version.match(/\/v?(\d+\.\d+\.\d+)/u)?.[1] ?? null;
+  const isMobile = version.endsWith('Mobile');
+  return (
+    compare(METAMASK_SUPPORTS_SNAPS_VERSION, currentVersion, '<=') && !isMobile
```
Comment (ardier16) [inline packages/connector/src/utils.ts]: why vice versa?

IMO `compare(currentVersion, METAMASK_SUPPORTS_SNAPS_VERSION, '>=')` is better

File: packages/connector/src/consts/general.ts [hunk 5]
```diff
@@ -1 +1,2 @@
 export const CORE_POLLING_INTERVAL = 5_000;
+export const METAMASK_SUPPORTS_SNAPS_VERSION = '11.5.0';
```
Comment (ardier16) [inline packages/connector/src/consts/general.ts]: mb `SUPPORTED_METAMASK_VERSION`?

**Review by ardier16 — APPROVED** [review]

**Pull request merged**